### ISSUE TITLE
Cloudlet EdgeRedirector JSON discrepancy for useRelativeUrl

### ISSUE THREAD

**Issue by hypnotisttom** [akamai/terraform-provider-akamai#328]
### Affected Resource(s)
Please list the resources as a list, for example:
- akamai_cloudlets_policy

### Expected Behavior
The policy in the UI should reflect the "none" value.

### Actual Behavior
When using a JSON policy, the policy in the UI reflected "copy_scheme_hostname" instead.

### Steps to Reproduce
When creating a cloudlet ER policy with this HCL:

```
resource "akamai_cloudlets_policy" "policy" {
  name          = var.policy_name
  cloudlet_code = var.cloudlet_code
  description   = var.description
  group_id      = var.property_group_id
  match_rules   = file("${path.module}/project-resources/cloudlets/edge-redirector/initial/match_rules.json")
}
```

Using this JSON:

```
[
  {
    "name": "rule1",
    "type": "erMatchRule",
    "statusCode": 301,
    "redirectURL": "https://www.example.com",
    "matchURL": "example.com",
    "useIncomingQueryString": false,
    "useIncomingSchemeAndHost": true,
    "useRelativeUrl": "none"
  },
  {
    "name": "rule2",
    "type": "erMatchRule",
    "matches": [
      {
        "matchType": "hostname",
        "matchValue": "example2.com",
        "matchOperator": "equals",
        "caseSensitive": true,
        "negate": false
      }
    ],
    "statusCode": 301,
    "redirectURL": "https://www.example.com",
    "useIncomingQueryString": false,
    "useIncomingSchemeAndHost": true,
    "useRelativeUrl": "none"
  },
  {
    "name": "rule3",
    "type": "erMatchRule",
    "matches": [
      {
        "matchType": "hostname",
        "matchValue": "example3b.com",
        "matchOperator": "equals",
        "caseSensitive": true,
        "negate": false
      }
    ],
    "statusCode": 301,
    "redirectURL": "https://www.example.com",
    "useIncomingQueryString": false,
    "useIncomingSchemeAndHost": true,
    "useRelativeUrl": "none"
  }
]
```

The UI is reflecting "Copy Protocol and Hostname" for useRelativeUrl instead of "None" even though the plan shows "none".

I do not know if this only affects that particular JSON key, as I haven't tested every key or every value for a key.

I have, however, attempted a similar policy with the HCL data source such as the below, and the value is set properly.

```
resource "akamai_cloudlets_policy" "policy" {
  name          = var.policy_name
  cloudlet_code = var.cloudlet_code
  description   = var.description
  group_id      = var.property_group_id
  match_rules   = data.akamai_cloudlets_edge_redirector_match_rule.example.json
}

data "akamai_cloudlets_edge_redirector_match_rule" "example" {
    match_rules {
        name = "rule"
        start = 1644865045
        end = 1645037845
        match_url = "example.com"
        redirect_url = "https://www.example.com"
        status_code = 301
        use_incoming_query_string = false
        use_relative_url = "none"
        matches {
            case_sensitive = false
            match_operator = "equals"
            match_type = "method"
            negate = false
            object_match_value {
                type = "simple"
                value = ["GET"]
            }
        }
    }
}
```

Our implementation does require the JSON rather than the HCL, as the JSON is provided by another team and then read in via CI/CD for the Terraform.

Based on this, I'm thinking there is some issue with the JSON implementation, although I'd be happy to be wrong.

Thanks,

Tom

**Comment by mgwoj** [comment]
Hello @hypnotisttom 

Thank you for reporting this issue. We will be evaluating it further.

Thanks,
Michal

**Comment by hypnotisttom** [comment]
After some additional research, it appears this issue is because "useIncomingSchemeAndHost" is set to true in the JSON with "useRelativeUrl" set to "none".  "useIncomingSchemeAndHost" is not in the HCL version.

The reason why it was in the JSON was because it was copied from the example on this page https://registry.terraform.io/providers/akamai/akamai/latest/docs/resources/cloudlets_policy.

I believe 1) the documentation should be updated, and 2) it's up to the devs if you want to catch/warn about the incompatibility between a "useRelativeUrl" set to "none" when "useIncomingSchemeAndHost" is set to true.

At this point I don't believe this to be a JSON provider issue.

Tom

**Comment by majakubiec** [comment]
Hi,
I'm happy to see that you have found the root of this issue.
The provider documentation example will be updated as part of the next terraform release.

As of displaying warnings, we decided not to do this because the resources are intended to be used with related datasources where we place validations.
We believe Cloudlets API behavior (including useIncomingSchemeAndHost and useRelativeUrl fields) should be documented in [cloudlets api docs](https://techdocs.akamai.com/cloudlets/v2/reference/post-policy-versions) for all that for some reason don't want to use datasources and want to use raw json instead. Unfortunately, api docs does not mention how useIncomingSchemeAndHost and useRelativeUrl effect one another, so I suggested the docs improvement to internal team and hopefully this will get fixed in near future. 

BR,
Mateusz
